### PR TITLE
Adding FSI Turek Benchmark to CoSim Examples

### PR DESCRIPTION
**Pull request by juancamarotti**
Adding one new FSI example to the CoSimulationApplication (FSI Turek example)

### CODE REVIEW

File: co_simulation/validation/fsi_turek_FSI2/source/ProjectParametersCSM.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        },
+        "convergence_criterion"              : "residual_criterion",
+        "displacement_relative_tolerance"    : 0.0001,
+        "displacement_absolute_tolerance"    : 1e-9,
+        "residual_relative_tolerance"        : 1e-6,
+        "residual_absolute_tolerance"        : 1e-8,
+        "max_iteration"                      : 10,
+        "linear_solver_settings"             : {
+            "solver_type" : "LinearSolversApplication.sparse_qr"
+        },
```
Comment (rubenzorrilla) [inline co_simulation/validation/fsi_turek_FSI2/source/ProjectParametersCSM.json:35]: I'll let the solver to take the fastest available one.

File: co_simulation/validation/fsi_turek_FSI2/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Turek benchmark - FSI2
```
Comment (rubenzorrilla) [inline co_simulation/validation/fsi_turek_FSI2/README.md:1]: This file is basically copying&pasting the one I did for the `FSIApplication` examples. I'd create a hyperlink to it in here.

File: co_simulation/validation/fsi_turek_FSI2/source/ProjectParametersCFD.json [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+            "domain_size"                  : 2,
+            "model_part_name"              : "FluidModelPart",
+            "model_import_settings"       : {
+                "input_type"     : "mdpa",
+                "input_filename" : "fsi_turek_FSI2_Fluid"
+            },
+            "echo_level"                  : 0,
+            "compute_reactions"           : true,
+            "formulation": {
+                "element_type": "vms",
```
[the remark this answers is not in the document]
Comment (rubenzorrilla) [reply co_simulation/validation/fsi_turek_FSI2/source/ProjectParametersCFD.json:29]: It is. The usage of `vms` is deprecated.

[the remark this answers is not in the document]
Comment (juancamarotti) [reply co_simulation/validation/fsi_turek_FSI2/source/ProjectParametersCFD.json:29]: Ok, I will change it

**Comment by matekelemen** [comment]
@juancamarotti can you plz update me about what the problem with the coupling convergence was?

**Comment by juancamarotti** [comment]
@matekelemen the ALE boundary conditions for the mesh motion solver were imposed incorrectly in the ProjectParametersCFD.json

**Comment by matekelemen** [comment]
alright, can you please give me a [diff](https://stackoverflow.com/a/40883538/12350793) on the setting that solved the issue?

**Comment by juancamarotti** [comment]
@rubenzorrilla I have already changed the fluid element formulation and it is working properly! Could you review it?

File: co_simulation/validation/fsi_turek_FSI2/source/ProjectParametersCoSim.json [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                        "model_part_name"   : "FluidModelPart.FluidNoSlipInterface2D_FluidInterface",
+                        "dimension" : 2,
+                        "variable_name" : "MESH_DISPLACEMENT"
+                    },
+                    "load" : {
+                        "model_part_name"   : "FluidModelPart.FluidNoSlipInterface2D_FluidInterface",
+                        "dimension" : 2,
+                        "variable_name" : "REACTION"
+                    },
+                    "velocity" : {
```
Comment (Igarizza) [inline co_simulation/validation/fsi_turek_FSI2/source/ProjectParametersCoSim.json:86]: I think you don't use velocity in this example. Can you delete it from settings?

**Comment by rubenzorrilla** [comment]
> @rubenzorrilla I have already changed the fluid element formulation and it is working properly! Could you review it?

Looks OK. I've nothing else to add.

**Comment by juancamarotti** [comment]
> > @rubenzorrilla I have already changed the fluid element formulation and it is working properly! Could you review it?
> 
> Looks OK. I've nothing else to add.

Could you please approve the pull request so that I can merge it to the master branch? Thank you very much in advance

**Comment by rubenzorrilla** [comment]
> > > @rubenzorrilla I have already changed the fluid element formulation and it is working properly! Could you review it?
> > 
> > 
> > Looks OK. I've nothing else to add.
> 
> Could you please approve the pull request so that I can merge it to the master branch? Thank you very much in advance

There is no branch protection in the Examples repo so you can merge whenever you want.

**Pull request merged**